### PR TITLE
`intersperseArray` utility

### PR DESCRIPTION
**Pull request by giraugh**

### CODE REVIEW

**Review by GRA0007 — APPROVED** [review]

File: lib/arrays/intersperseArray.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  it('works for example 1', () => {
+    const inter = intersperseArray([[1, 2, 3], [4, 5, 6], [7, 8, 9]], [])
+    expect(inter).toEqual([[1, 2, 3], [], [4, 5, 6], [], [7, 8, 9]])
+  })
+
+  it('works for example 2', () => {
+    const inter = intersperseArray(['alpha', 'beta', 'gamma'], '!')
+    expect(inter).toEqual(['alpha', '!', 'beta', '!', 'gamma'])
+  })
```
Comment (GRA0007) [inline lib/arrays/intersperseArray.ts:35]: Needs some more edge case tests

```suggestion
  it('works for example 1', () => {
    const inter = intersperseArray([[1, 2, 3], [4, 5, 6], [7, 8, 9]], [])
    expect(inter).toEqual([[1, 2, 3], [], [4, 5, 6], [], [7, 8, 9]])
  })

  it('works for example 2', () => {
    const inter = intersperseArray(['alpha', 'beta', 'gamma'], '!')
    expect(inter).toEqual(['alpha', '!', 'beta', '!', 'gamma'])
  })
  
  it('does nothing to an empty array', () => {
    expect(intersperseArray([] as string[], 'hello')).toEqual([])
  })
  
  it('does nothing to an array with one element', () => {
    expect(intersperseArray(['suave'], '$')).toEqual(['suave'])
  })
```
